config/tf_modules/gcp-k8s-service/k8s-service/templates/deployment.yaml

```diff
@@ -42,31 +42,36 @@ spec:
           ports:
             - name: main
               protocol: TCP
-            {{- if hasKey .Values.port "http" }}
+              {{- if hasKey .Values.port "http" }}
               containerPort: {{ .Values.port.http }}
-            {{- end }}
-            {{- if hasKey .Values.port "tcp" }}
+              {{- end }}
+              {{- if hasKey .Values.port "tcp" }}
               containerPort: {{ .Values.port.tcp }}
-            {{- end }}
-            {{- if hasKey .Values.port "grpc" }}
+              {{- end }}
+              {{- if hasKey .Values.port "grpc" }}
               containerPort: {{ .Values.port.grpc }}
-            {{- end }}
+          {{- end }}
           env:
             - name: RDS_CA_PATH # This is the path to the public key for the docdb tls
               value: "/config/rds_ca.pem"
-            {{- if or (.Values.envVars) (.Values.secrets) }}
             {{ range .Values.envVars }}
             - name: {{ .name | quote }}
               value: {{ .value | quote }}
             {{ end }}
-            {{ range $val := .Values.secrets }}
+            {{ range $val := .Values.linkSecrets }}
             - name: {{ $val.name | quote }}
               valueFrom:
                 secretKeyRef:
                   name: secret
                   key: {{ $val.name | quote }}
             {{ end }}
-            {{- end }}
+            {{ range $val := .Values.manualSecrets }}
+            - name: {{ $val | quote }}
```

Comment:
Shouldn't this be `$val.name` like above?

Reply:
No in manual secrets we only pass the list of secret names because there is no need for the helm chart to know the values.